client/scala/src/main/scala/ai/verta/blobs/dataset/FileMetadata.scala

```diff
@@ -5,11 +5,19 @@ package ai.verta.blobs.dataset
  *  @param md5 MD5 hash of the file
  *  @param path path of the file
  *  @param size size of the file
+ *  @param versionId: (optional) versionId of the file. Only exists for S3 files
  */
-class FileMetadata(val lastModified: BigInt, val md5: String, val path: String, val size: BigInt) {
+class FileMetadata(
+  val lastModified: BigInt,
+  val md5: String,
+  val path: String,
+  val size: BigInt,
+  val versionId: Option[String] = None
+) {
   override def equals(other: Any) = other match {
     case other: FileMetadata => lastModified == other.lastModified &&
-    md5 == other.md5 && path == other.path && size == other.size
+    md5 == other.md5 && path == other.path && size == other.size &&
+    ((versionId.isEmpty && other.versionId.isEmpty) || (versionId.get == other.versionId.get))
```

Comment:
This will raise an exception if one of them is empty and the other isn't.

Reply:
Oops I was dumb. Forgot to add the condition that both are defined.

Reply:
Fixed.